li-utils/src/main/javaPegasus/com/linkedin/common/urn/AzkabanJobUrn.java

```diff
@@ -10,38 +10,53 @@
 
   public static final String ENTITY_TYPE = "azkabanJob";
 
-  private static final String CONTENT_FORMAT = "(%s,%s)";
-
-  private final AzkabanFlowUrn flowEntity;
-
-  private final String jobIdEntity;
+  private final AzkabanFlowUrn _flow;
+  private final String _jobId;
 
   public AzkabanJobUrn(AzkabanFlowUrn flow, String jobId) {
-    super(ENTITY_TYPE, String.format(CONTENT_FORMAT, flow.toString(), jobId));
-    this.flowEntity = flow;
-    this.jobIdEntity = jobId;
+    super(ENTITY_TYPE, TupleKey.create(flow, jobId));
+    this._flow = flow;
+    this._jobId = jobId;
   }
 
-  public AzkabanFlowUrn getFlowEntity() {
-    return flowEntity;
+  public AzkabanFlowUrn getFlow() {
+    return _flow;
   }
 
-  public String getJobIdEntity() {
-    return jobIdEntity;
+  public String getJobId() {
+    return _jobId;
   }
 
   public static AzkabanJobUrn createFromString(String rawUrn) throws URISyntaxException {
-    String content = new Urn(rawUrn).getContent();
-    String flowParts = content.substring(1, content.lastIndexOf(",") + 1);
-    String[] parts = content.substring(1, content.length() - 1).split(",");
-    return new AzkabanJobUrn(AzkabanFlowUrn.createFromString(flowParts), parts[3]);
+    return createFromUrn(Urn.createFromString(rawUrn));
+  }
+
+  public static AzkabanJobUrn createFromUrn(Urn urn) throws URISyntaxException {
+    if (!"li".equals(urn.getNamespace())) {
+      throw new URISyntaxException(urn.toString(), "Urn namespace type should be 'li'.");
+    } else if (!ENTITY_TYPE.equals(urn.getEntityType())) {
+      throw new URISyntaxException(urn.toString(), "Urn entity type should be 'azkabanJob'.");
```

Comment:
this validation seems to be common across all entity urns. We can move this to a util function?

Reply:
No.

1) The entity type is different in each.
2) The amount of parts in the tuple are different.
3) The constructors are all different.
4) **Most important** Ideally we can just code gen this instead, like we do internally.

Reply:
I'm suggesting that we move the first two checks to a separate method. `urn.getNamespace()` and `urn.getEntityType()` should be valid for any urn? You can always pass the entity type in the util function.

Reply:
Ah, ok, I didn't realize there were getters for those. I think there's two approaches:

1) Add a util function to GMA that we'll probably later delete when we do code gen ("duplicate" code in code gen is probably fine and even preferable?).
2) Keep as is since eventually all this will be deleted (generated) anyway.

I'm fine with either, no strong feelings either way.

Reply:
When do you think will we be able to code gen this? If it's in near future, we can leave this as it is. If not, we might want to add the util function.

Reply:
We should probably do it asap :) It is a pain to maintain these lol